libsolidity/ast/AST.h

```diff
@@ -1143,6 +1144,46 @@ class EventDefinition: public CallableDeclaration, public StructurallyDocumented
 	bool m_anonymous = false;
 };
 
+/**
+ * Definition of an error type usable in ``revert(MyError(x))``, ``require(condition, MyError(x))``
+ * and ``catch MyError(_x)``.
+ */
+class ErrorDefinition: public CallableDeclaration, public StructurallyDocumented, public ScopeOpener
+{
+public:
+	ErrorDefinition(
+		int64_t _id,
+		SourceLocation const& _location,
+		ASTPointer<ASTString> const& _name,
+		ASTPointer<StructuredDocumentation> const& _documentation,
+		ASTPointer<ParameterList> const& _parameters
+	):
+		CallableDeclaration(_id, _location, _name, Visibility::Default, _parameters),
+		StructurallyDocumented(_documentation)
+	{
+	}
+
+	void accept(ASTVisitor& _visitor) override;
+	void accept(ASTConstVisitor& _visitor) const override;
+
+	TypePointer type() const override;
+
+	FunctionTypePointer functionType(bool /*_internal*/) const override;
```

Comment:
It's a declaration so why not:
```suggestion
	FunctionTypePointer functionType(bool _internal) const override;
```

Reply:
Fixed.